service/controller/v5/resource/instance/deployment.go

```diff
@@ -2,17 +2,43 @@ package instance
 
 import (
 	"context"
+	"encoding/base64"
 
 	azureresource "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/azure-storage-blob-go/azblob"
 	"github.com/Azure/go-autorest/autorest/to"
 	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/microerror"
 
+	"github.com/giantswarm/azure-operator/service/controller/v5/blobclient"
 	"github.com/giantswarm/azure-operator/service/controller/v5/controllercontext"
 	"github.com/giantswarm/azure-operator/service/controller/v5/key"
+	"github.com/giantswarm/azure-operator/service/controller/v5/templates"
 )
 
 func (r Resource) newDeployment(ctx context.Context, obj providerv1alpha1.AzureConfig, overwrites map[string]interface{}) (azureresource.Deployment, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return azureresource.Deployment{}, microerror.Mask(err)
+	}
+	err = cc.Validate()
+	if err != nil {
+		return azureresource.Deployment{}, microerror.Mask(err)
+	}
+
+	cloudConfigURLs := []string{
+		key.BlobName(obj, key.PrefixMaster()),
+		key.BlobName(obj, key.PrefixWorker()),
+	}
+
+	for _, key := range cloudConfigURLs {
+		blobURL := cc.ContainerURL.NewBlockBlobURL(key)
+		_, err := blobURL.GetProperties(ctx, azblob.BlobAccessConditions{})
```

Comment:
Why is this line required?

Reply:
If blob is not ready - we return error that instance resource creation should be canceled?

Reply:
Ack.

Reply:
Would be good to add a comment.